hw/ip/sram_ctrl/doc/_index.md

```diff
@@ -103,35 +121,55 @@ This mechanism is part of the [life cycle]({{< relref "hw/ip/lc_ctrl/doc" >}}) s
 
 Another CTR mode augmentation that is aimed at breaking the linear address space is SRAM address scrambling.
-The same two-layer S&P network that is used for byte diffusion is leveraged to non-linearly remap the SRAM address as shown in the block diagram above.
-As opposed to the byte diffusion S&P networks, this particular address scrambling network additionally XOR's in a nonce that has the same width as the address.
+The same two-layer S&P network that is used for intra-word diffusion is leveraged to non-linearly remap the SRAM address as shown in the block diagram above.
+As opposed to the intra-word diffusion S&P networks, this particular address scrambling network additionally XOR's in a nonce that has the same width as the address.
 
 ### Integrity Error Handling
 
-The `prim_ram_1p_scr` primitive contains an input to indicate whether upstream logic has detected an integrity error (The generation of the integrity error is determined by system integration).
-When an integrity error is encountered, the prim memory module does the following:
-*  Latch the error condition until reset.
+When an integrity error is encountered, the `sram_ctrl` will latch the integrity error send out a `fatal_bus_integ_error` until the next reset (the generation of the integrity error is determined by system integration).
+In addition, this the latched error condition is fed into the `prim_ram_1p_scr` primitive via a dedicated input, causing the scrambling primitive to do the following:
 *  Reverse the nonce used during the address and CTR scrambling.
 *  Disallow any transaction (read or write) on the actual memory macro.
 
 This behavior, combined with other top level defenses, form a multi-layered defense when integrity errors are seen in the system.
 
+### LFSR Initialization Feature
+
+Note that since the scrambling device uses a block cipher in CTR mode, it is undesirable to initialize the memory with all-zeros from a security perspective, as that would reveal the XOR keystream.
+To this end, the `sram_ctrl` contains an LFSR-based initialization mechanism that overwrites the the entire memory with pseudorandom data.
+
+Initialization can be triggered via the {{< regref "CTRL.INIT" >}} CSR, and once triggered, the LFSR is first re-seeded with the nonce that has been fetched together with the scrambling key.
+Then, the memory is initialized with pseudorandom data pulled from the LFSR.
+For each pseudorandom 32bit word, the initialization mechanism computes the corresponding integrity bits and writes both the data and integrity bits (39bit total) through the scrambling device using the most recently obtained scrambling key.
+
+If SW triggers the scrambling key renewal and LFSR initialization at the same time (i.e., with the same CSR write operation), the LFSR initialization will be stalled until a new scrambling has been obtained.
```

Comment:
are you thinking like a separate `prim_lfsr_integ_init` primitive, or rather `prim_ram_1p_scr_init` wrapper?

Reply:
o no, i meant literally something like `sys_intg_enc` and `sys_intg_dec`, whose sole purpose is to wrap up the encoder/decoder.

That way if we find something more efficient in the future, i can just swap out what's underneath, instead of hunting down all the usage of ecc hehe. Completely unrelated to this PR, it was just a thought.

Reply:
ok sounds good, I'll do that.

Reply:
done here: https://github.com/lowRISC/opentitan/pull/7631